docs(en): optional step for custom gh-pages domain (#1110)
withastro/docs: src/pages/en/guides/deploy/github.md

    This workflow uses the `npm ci` command by default. You must include a `package-lock.json` file in your repository for this to work. To generate one, run `npm i` in your terminal and commit the resulting lock file.
    :::

3. Commit the new workflow file and push it to GitHub.  

4. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.  

5. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.  
  

> :::tip
> If you want a custom domain rather than `user.github.io`, follow [the official Github Guide for configuring domains](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain) and make sure to place the `CNAME` file into your `public` folder.
> :::

Reviewer: ```suggestion
:::tip
If you want a custom domain rather than `user.github.io`, create a file named `CNAME` (no extension) in your `public` folder. The content of the file should be just your Domain like `example.com` or `subdomain.example.com`
:::
```

Reviewer 2: I will point out that you made this change in the first place because you thought we didn't need instructions here since there was a link to GH's instructions! 

Even if GH's instructions are confusing, having a link to them does provide help to the reader, especially being the official documentation for what they are trying to achieve. I think this takes too much away from the reader.

Reviewer 3: I think we might be diving a little off the purpose of the deploy guide. My understanding was that this guide was to help the user publish their project online. 

I like the idea of having an aside or tip to point out further configuration to the respective documentation guides.

Reviewer: The problem with the GH Docs are that they just don't make it clear that much because they use a different approach then we have here. The problem originally was more that it was a lot to read before going on with the main steps. That's why I trimmed it down.

    This workflow uses the `npm ci` command by default. You must include a `package-lock.json` file in your repository for this to work. To generate one, run `npm i` in your terminal and commit the resulting lock file.
    :::

3. Commit the new workflow file and push it to GitHub.  

4. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.  

5. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.  
  
Your site should now be published! When you push changes to your Astro project’s repository, the GitHub Action will automatically deploy them for you.

:::tip[set up a custom domain]
You can optionally set up a custom domain by adding the following `./public/CNAME` file to your project: 

```txt title="public/CNAME"
sub.mydomain.com
```

> `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).   

Reviewer 2: ```suggestion
This will replace `user.github.io` with your custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).   
```

Reviewer: That's good :+1:

> `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).   

Reviewer: Maybe add a notice that Step 5 is the DNS Setup Step. Because the first 4 are already covered by Astros Docs (and works a bit differently)

Reviewer 2: Hmm... step 5 hasn't changed with this new addition re: custom domain, so I think it probably is unaffected by adding new content here? I'm tempted to keep it as it was a previously existing step.

Reviewer: I meant the GH Docs...but it's merged now anyways so...

    :::tip
    See [the GitHub Pages Action documentation](https://github.com/marketplace/actions/github-pages-action) for different ways you can configure the final “Deploy to GitHub Pages” step.
    :::
3. (Optionally) Set up a custom domain by adding the following `./public/CNAME` file to your project: 

> ```
> sub.mydomain.com
> ````
> `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).     
> 

Reviewer 4: We should indent these items by four spaces to make sure they get nested correctly under item 3. I’ve also added a filename to the example CNAME code block.

````suggestion

    ```txt title="public/CNAME"
    sub.mydomain.com
    ```

    `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).     

````

>     :::tip
>     See [the GitHub Pages Action documentation](https://github.com/marketplace/actions/github-pages-action) for different ways you can configure the final “Deploy to GitHub Pages” step.
>     :::

Reviewer: Forgot one
```suggestion
```

> 3. (Optionally) Set up a custom domain by adding the following `./public/CNAME` file to your project: 

Reviewer: ```suggestion
```

>     ```txt title="public/CNAME"
>     sub.mydomain.com
>     ```
> 
>     `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).     
> 
> 4. Commit the new workflow file and push it to GitHub.  
> 
> 5. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.  
> 
> 6. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.  
>   
> 
> 7. (Optionally) Set up a custom domain by adding the following `./public/CNAME` file to your project:   
> 
> 
>     ```txt title="public/CNAME"
>     sub.mydomain.com
>     ```
> 
>     `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).     

Reviewer: ```suggestion
3. Commit the new workflow file and push it to GitHub.  

4. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.  

5. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.  
  

:::tip
If you want a custom domain rather than `user.github.io`, follow [the official Github Guide for configuring domains](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain) and make sure to place the `CNAME` file into your `public` folder.
:::
```

3. Commit the new workflow file and push it to GitHub.
4. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.
5. Choose the `gh-pages` branch and the `"/" (root)` folder as the **Source** of your site and press **Save**.

> 6. (Optionally) Setup a custom domain by adding `./public/CNAME` file with `sub.mydomain.com` as content.  
>     `actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment.  
>     Don't forget to [setup DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).
> 

Reviewer 2: ``````suggestion
6. Optional: Set up a custom domain by adding the following `./public/CNAME` file to your project: 
```
sub.mydomain.com
````
`actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment. Don't forget to also [configure DNS for your domain provider](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain).     
``````
Hey @JulianCataldo thanks so much for stepping up for this! 🙌 

Looks great, I'm just thinking of maybe a way to separate a direct, explicit instruction from the under-the-hood explanation as clearly as possible. And, when a new file must be created, if we can figure out a way to do it sensibly and make it look nice, we do like to provide a code sample. 

Also, line breaks to start a new sentence without a blank line between them don't look the best in our lists on the Docs site, I find. So, we try to structure things where a blank line between paragraphs makes sense/looks good, or keep everything in one paragraph (again, if it doesn't look terrible). Welcome to Docs: not just getting the content right, but sometimes changing content to fit the visuals!

I can't see a preview of this while I'm just reviewing the file diff on GitHub's website, so this isn't a set in stone suggestion for change. Just wondering if some kind of tweaking like this might give us a cleaner look on the site. What do you think? Do you mind experimenting?

Author: You're welcome :)

Under-the-hood things might be totally superfluous.  
People just want to get the job done ;p
OK for a second version with an attempt at better spacings.

About "(Optionally)" I've copied this from another place in docs, for coherence sake. BTW, I originally came with "(Optional)" as a first guess.

Also, I think that:

> Blabla: some blaba: more blah.

(Two colons in the same sentence) is syntactically incorrect, in English / French…

![Screenshot 2022-07-25 at 4 00 42](https://user-images.githubusercontent.com/603498/180678788-4b698ef9-8a60-4bd0-a7c8-8b6265d5ee48.png)

Reviewer 2: Yes, if adding the file should really be done earlier, then I think moving it up sounds good! 

How important/helpful would you rank this info:  "`actions-gh-pages` will detect it in `./dist` after build, and will replace `user.github.io` with the correct custom destination domain for your deployment environment."  This is what I was wondering about re: maybe unnecessary info for what are otherwise pretty short steps? 

And yes, let's keep the optionally consistent then. Those integrations docs were all written separately, and the content is mirrored in from an external source on the astro repo, so I sometimes forget to consider their style, too. Great catch!

Author: Mhhh, I'm not quite accustomed to this "hot-edit" workflow, I can't find a way to (re)edit the current PR / concerned files 😅

Reviewer 2: Ooh, one thing I do is just, on this PR on the GitHub.com itself, is there's a tab at the top for "Files Changed."  (Right now, we're in "Conversation.")

If you click to see the files changed, then as you hover over lines along the gutter, you should see a big PLUS sign visible. If you click and hold and drag down, you can select multiple lines to edit.  It will look like you're just making a comment at first, but if you press the icon that looks like a document with a plus sign in the middle, you'll actually be able to "suggest" edits to what's there. So, you don't get a nice preview, but you can literally change files that way and make "suggestions" and evne commit them that way.

Let me know if trying that doesn't work for you!
